testtools/UART_interface/rpi_uart_interface.py

```diff
@@ -0,0 +1,197 @@
+        # may want special opt handling for raspi commands
+
+        if ser.readable():
+            output = ser.readline(ser.in_waiting)
+            output = output.decode(encoding='utf-8', errors='ignore').split()
+
+            # check_sdk_errors(output)
+            check_test_failures(output)
+            print(output)
+            try:
+                # File must exist to write to it
+                file.writelines(output)
+            except:
+                pass
+
+            return output
+        else:
+            try:
+                time.sleep(2)
+                ser.open()
+                self.serial_read(ser, message, file, first_read=True)
+            except:
+                return False
+
+    # Note: the buffer size on the mxchip appears to be 128 Bytes.
+    def write_read(self, ser, input_file, output_file):
+        session_start = time.time()
+        if input_file:
+            # set wait between read/write
+            wait = (serial_settings.bits_to_cache/serial_settings.baud_rate)
+            with open(input_file) as fp:
+                # initialize input/output_file
+                line = fp.readline()
+                f = open(output_file, 'w+') # Output file
+                while line:
+                    # time.sleep(.1)
+                    # Print only instruction, not secret content
+                    if line.split():
+                        print("Sending %s" %line.split()[0])
+
+                    # Attempt to write to serial port
+                    if not self.serial_write(ser, line, f):
+                        print("Failed to write to serial port, please diagnose connection.")
+                        f.close()
+                        break
+                    time.sleep(1)
+
+                    # Attempt to read serial port
+                    output = self.serial_read(ser, line, f, first_read=True)
+
+                    while(output):
+                        time.sleep(wait)
+                        output = self.serial_read(ser, line, f)
+                    line = fp.readline()
+
+                # read any trailing output, save to file
+                if serial_settings.test_timeout:
+                    while((time.time() - session_start) < serial_settings.test_timeout):
+                    # while (ser.in_waiting):
+                        time.sleep(.2)
+                        output = ser.readline(ser.in_waiting)#self.serial_read(ser, line, f)
+                        output = output.decode(encoding='utf-8', errors='ignore')
+                        check_test_failures(output)
+
+                        if len(output) > 4:
+                            print(output)
+
+                        #for now we can assume one test suite is run
+                        if " tests run" in output:
+                            break
+                    # reset after every test
+                    ser.write(bytearray("sudo reboot\n".encode("ascii")))
```

Comment:
>ser.write(bytearray("sudo reboot\n".encode("ascii"))) [](start = 20, length = 53)

We could add an "rebootAfterTest" input flag for the whole script that adds a condition around this line. That way it isn't forced

Reply:
Added it as a command line option.